lib/explorer/data_frame.ex

```diff
@@ -682,34 +690,40 @@ defmodule Explorer.DataFrame do
 
     backend = backend_from_options!(backend_opts)
 
-    backend.from_parquet(
-      normalise_entry(filename, opts[:config]),
-      opts[:max_rows],
-      to_columns_for_io(opts[:columns])
-    )
+    with {:ok, entry} <- normalise_entry(filename, opts[:config]) do
+      backend.from_parquet(
+        entry,
+        opts[:max_rows],
+        to_columns_for_io(opts[:columns])
+      )
+    end
   end
 
   defp normalise_entry(%_{} = entry, config) when config != nil do
-    raise ArgumentError,
-          ":config key is only supported when the argument is a string, got #{inspect(entry)} with config #{inspect(config)}"
+    {:error,
+     ArgumentError.message(
+       ":config key is only supported when the argument is a string, got #{inspect(entry)} with config #{inspect(config)}"
+     )}
   end
 
-  defp normalise_entry(%Local.Entry{} = entry, nil), do: entry
-  defp normalise_entry(%S3.Entry{config: %S3.Config{}} = entry, nil), do: entry
+  defp normalise_entry(%Local.Entry{} = entry, nil), do: {:ok, entry}
+  defp normalise_entry(%S3.Entry{config: %S3.Config{}} = entry, nil), do: {:ok, entry}
 
   defp normalise_entry("s3://" <> _rest = entry, config) do
     config = s3_config(config)
-    %S3.Entry{url: entry, config: config}
+    {:ok, %S3.Entry{url: entry, config: config}}
   end
 
-  defp normalise_entry("file://" <> path, _config), do: %Local.Entry{path: path}
+  defp normalise_entry("file://" <> path, _config), do: {:ok, %Local.Entry{path: path}}
 
   defp normalise_entry(filepath, _config) when is_binary(filepath) do
     if File.exists?(filepath) do
-      %Local.Entry{path: filepath}
+      {:ok, %Local.Entry{path: filepath}}
```

Comment:
What if we always return it as `%Local.Entry{path: filepath}`? Then if it doesn't exist, Polars will fail to read anyway and say the file does not exist. This way we don't need to fiddle with ArgumentError below. Thoughts?

Reply:
But you mean, remove the result from the `normalise_entry/2` entirely? Or just always return `{:ok, %Local.Entry{path: filepath}}` for this function clause?

Reply:
Don't check if the file exists, alwaus return `{:ok, ...}` for the function clause.